src/browser/modules/D3Visualization/components/Explorer.tsx

```diff
@@ -121,13 +135,18 @@ export class ExplorerComponent extends Component<any, ExplorerComponentState> {
   }
 
   onGraphModelChange(stats: any) {
-    this.setState({ stats: stats })
+    this.setState((oldState: any) => ({
```

Comment:
could this be made simpler? Also, are we sure we won't be keeping around old/stale stats by spreading the old ones like this?

Reply:
Fixed